### ISSUE TITLE
0.26.0 Disable debug log `statusUpdateHook`

### ISSUE THREAD

**Issue by yuriydzobak** [actions-runner-controller/actions-runner-controller#1939]
### Checks

- [X] I've already read https://github.com/actions-runner-controller/actions-runner-controller/blob/master/TROUBLESHOOTING.md and I'm sure my issue is not covered in the troubleshooting guide.
- [X] I'm not using a custom entrypoint in my runner image

### Controller Version

0.26.0

### Helm Chart Version

0.21.0

### Deployment Method

Helm

### cert-manager installation

yes

### Checks

- [X] This isn't a question or user support case (For Q&A and community support, go to [Discussions](https://github.com/actions-runner-controller/actions-runner-controller/discussions). It might also be a good idea to contract with any of contributors and maintainers if your business is so critical and therefore you need priority support
- [X] I've read [releasenotes](https://github.com/actions-runner-controller/actions-runner-controller/tree/master/docs/releasenotes) before submitting this issue and I'm sure it's not due to any recently-introduced backward-incompatible changes
- [X] My actions-runner-controller version (v0.x.y) does support the feature
- [X] I've already upgraded ARC (including the CRDs, see charts/actions-runner-controller/docs/UPGRADING.md for details) to the latest and it didn't fix the issue
- [X] I've migrated to the workflow job webhook event (if you using webhook driven scaling)

### Resource Definitions

```yaml
apiVersion: actions.summerwind.dev/v1alpha1
kind: RunnerDeployment
metadata:
  name: gha-self-hosted-runner
  namespace: ci-runners
spec:
  template:
    metadata:
      labels:
        disalow-privileged-containers/enable: "false"
        app.kubernetes.io/name: gha-self-hosted-runner
      annotations:
        iam.amazonaws.com/role: lf_gha_assumer
        cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
    spec:
      env:
      # See: https://github.com/actions-runner-controller/actions-runner-controller/blob/master/TROUBLESHOOTING.md#delay-in-jobs-being-allocated-to-runners
      - name: DISABLE_RUNNER_UPDATE
        value: "true"
      - name: LOG_DEBUG_DISABLED
         value: "true"        
      dockerdWithinRunnerContainer: false
      dockerdContainerResources:
        limits:
          memory: "2Gi"
        requests:
          cpu: "150m"
          memory: "256Mi"
      ephemeral: true
      resources:
        requests:
          memory: "128Mi"
          cpu: "200m"
        limits:
          memory: "2Gi"
      imagePullPolicy: IfNotPresent
      organization: myorg
      serviceAccountName: github-action
      nodeSelector:
        group-name: ci-general-group
      tolerations:
      - effect: NoSchedule
        key: node-role.kubernetes.io/ci-general
      topologySpreadConstraints:
      - maxSkew: 1
        topologyKey: kubernetes.io/hostname
        whenUnsatisfiable: ScheduleAnyway
        labelSelector:
          matchLabels:
            app.kubernetes.io/name: gha-self-hosted-runner
      volumes:
      - name: docker-login
        secret:
          secretName: docker-login
          items:
          - key: ".dockerconfigjson"
            path: "config.json"
      volumeMounts:
      - mountPath: /home/runner/.docker/config.json
        name: docker-login
        subPath: config.json
      dockerVolumeMounts:
      - mountPath: /root/.docker/config.json
        name: docker-login
        subPath: config.json
```

### To Reproduce

1. Enable the feature in helm chart -> https://github.com/actions-runner-controller/actions-runner-controller/blob/master/docs/releasenotes/0.26.md#enhancement--more-granular-and-real-time-runner-statuses
```yaml
    runner:
      statusUpdateHook:
        enabled: true  
```

### Describe the bug

The runner's displays debug info and that has sensitive data(TOKEN, etc)
I can't disable it, I tried to add ENV variable to self-hosted POD but it didn't help -> https://github.com/actions-runner-controller/actions-runner-controller/blob/5298c6ea298152d7b6422f3c75e83ed9ea019572/runner/logger.bash#L25

```
 - name: LOG_DEBUG_DISABLED
  value: "true"
```

Example
![image](https://user-images.githubusercontent.com/70581130/196851610-91a2397b-c636-4b08-a716-a80e9fbecf52.png)
And anybody can see token
![image](https://user-images.githubusercontent.com/70581130/196853476-4d847cfb-2661-425c-9ec5-aa8204d75223.png)

### Describe the expected behavior

disable debug INFO

### Whole Runner Pod Logs

```shell
{
  "apiVersion": "actions.summerwind.dev/v1alpha1",
  "kind": "Runner",
  "metadata": {
    "annotations": {
      "cluster-autoscaler.kubernetes.io/safe-to-evict": "false",
      "iam.amazonaws.com/role": "lf_gha_assumer",
      "sync-time": "2022-10-20T02:11:00Z"
    },
    "creationTimestamp": "2022-10-20T02:11:00Z",
    "finalizers": [
      "runner.actions.summerwind.dev"
    ],
    "generateName": "gha-self-hosted-runner-4kp8x-",
    "generation": 1,
    "labels": {
      "app.kubernetes.io/name": "gha-self-hosted-runner",
      "disalow-privileged-containers/enable": "false",
      "runner-deployment-name": "gha-self-hosted-runner",
      "runner-template-hash": "6776bdd8db"
    },
    "managedFields": [
      {
        "apiVersion": "actions.summerwind.dev/v1alpha1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:metadata": {
            "f:annotations": {
              ".": {},
              "f:cluster-autoscaler.kubernetes.io/safe-to-evict": {},
              "f:iam.amazonaws.com/role": {},
              "f:sync-time": {}
            },
            "f:finalizers": {
              ".": {},
              "v:\"runner.actions.summerwind.dev\"": {}
            },
            "f:generateName": {},
            "f:labels": {
              ".": {},
              "f:app.kubernetes.io/name": {},
              "f:disalow-privileged-containers/enable": {},
              "f:runner-deployment-name": {},
              "f:runner-template-hash": {}
            },
            "f:ownerReferences": {
              ".": {},
              "k:{\"uid\":\"a035e991-1c56-4bf9-a387-b19252ca0d66\"}": {}
            }
          },
          "f:spec": {
            ".": {},
            "f:dockerVolumeMounts": {},
            "f:dockerdContainerResources": {
              ".": {},
              "f:limits": {
                ".": {},
                "f:memory": {}
              },
              "f:requests": {
                ".": {},
                "f:cpu": {},
                "f:memory": {}
              }
            },
            "f:dockerdWithinRunnerContainer": {},
            "f:env": {},
            "f:ephemeral": {},
            "f:image": {},
            "f:imagePullPolicy": {},
            "f:nodeSelector": {
              ".": {},
              "f:group-name": {}
            },
            "f:organization": {},
            "f:resources": {
              ".": {},
              "f:limits": {
                ".": {},
                "f:memory": {}
              },
              "f:requests": {
                ".": {},
                "f:cpu": {},
                "f:memory": {}
              }
            },
            "f:serviceAccountName": {},
            "f:tolerations": {},
            "f:topologySpreadConstraints": {},
            "f:volumeMounts": {},
            "f:volumes": {}
          }
        },
        "manager": "manager",
        "operation": "Update",
        "time": "2022-10-20T02:11:00Z"
      },
      {
        "apiVersion": "actions.summerwind.dev/v1alpha1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:status": {
            ".": {},
            "f:registration": {
              ".": {},
              "f:expiresAt": {},
              "f:organization": {},
              "f:token": {}
            }
          }
        },
        "manager": "manager",
        "operation": "Update",
        "subresource": "status",
        "time": "2022-10-20T02:11:00Z"
      },
      {
        "apiVersion": "actions.summerwind.dev/v1alpha1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:status": {
            "f:message": {},
            "f:phase": {}
          }
        },
        "manager": "curl",
        "operation": "Update",
        "subresource": "status",
        "time": "2022-10-20T02:13:45Z"
      }
    ],
    "name": "gha-self-hosted-runner-4kp8x-f426w",
    "namespace": "ci-runners",
    "ownerReferences": [
      {
        "apiVersion": "actions.summerwind.dev/v1alpha1",
        "blockOwnerDeletion": true,
        "controller": true,
        "kind": "RunnerReplicaSet",
        "name": "gha-self-hosted-runner-4kp8x",
        "uid": "a035e991-1c56-4bf9-a387-b19252ca0d66"
      }
    ],
    "resourceVersion": "929416318",
    "uid": "a685e75e-56e5-403c-94cf-d9eec37cbc5c"
  },
  "spec": {
    "dockerVolumeMounts": [
      {
        "mountPath": "/root/.docker/config.json",
        "name": "docker-login",
        "subPath": "config.json"
      }
    ],
    "dockerdContainerResources": {
      "limits": {
        "memory": "2Gi"
      },
      "requests": {
        "cpu": "150m",
        "memory": "256Mi"
      }
    },
    "dockerdWithinRunnerContainer": false,
    "env": [
      {
        "name": "DISABLE_RUNNER_UPDATE",
        "value": "true"
      }
      {
        "name": "LOG_DEBUG_DISABLED",
        "value": "true"
      }
    ],
    "ephemeral": true,
    "image": "summerwind/actions-runner:v2.298.2-ubuntu-20.04",
    "imagePullPolicy": "IfNotPresent",
    "nodeSelector": {
      "group-name": "ci-general-group"
    },
    "organization": "myorg",
    "resources": {
      "limits": {
        "memory": "2Gi"
      },
      "requests": {
        "cpu": "200m",
        "memory": "128Mi"
      }
    },
    "serviceAccountName": "github-action",
    "tolerations": [
      {
        "effect": "NoSchedule",
        "key": "node-role.kubernetes.io/ci-general"
      }
    ],
    "topologySpreadConstraints": [
      {
        "labelSelector": {
          "matchLabels": {
            "app.kubernetes.io/name": "gha-self-hosted-runner"
          }
        },
        "maxSkew": 1,
        "topologyKey": "kubernetes.io/hostname",
        "whenUnsatisfiable": "ScheduleAnyway"
      }
    ],
    "volumeMounts": [
      {
        "mountPath": "/home/runner/.docker/config.json",
        "name": "docker-login",
        "subPath": "config.json"
      }
    ],
    "volumes": [
      {
        "name": "docker-login",
        "secret": {
          "items": [
            {
              "key": ".dockerconfigjson",
              "path": "config.json"
            }
          ],
          "secretName": "docker-login"
        }
      }
    ]
  },
  "status": {
    "message": "",
    "phase": "Registering",
    "registration": {
      "expiresAt": "2022-10-20T03:00:45Z",
      "organization": "myorg",
      "token": "OMIT*****KC453AVPNFXHG5DBNRWGC5DJN5XF62LEZYAQHIKTWFUW443UMFWGYYLUNFXW4X3UPFYGLN2JNZ2GKZ3SMF2GS33OJFXHG5DBNRWGC5DJN5XA"
    }
  }
2022-10-20 02:19:35.316  DEBUG --- Configuring the runner.
}
--------------------------------------------------------------------------------
|        ____ _ _   _   _       _          _        _   _                      |
|       / ___(_) |_| | | |_   _| |__      / \   ___| |_(_) ___  _ __  ___      |
|      | |  _| | __| |_| | | | | '_ \    / _ \ / __| __| |/ _ \| '_ \/ __|     |
|      | |_| | | |_|  _  | |_| | |_) |  / ___ \ (__| |_| | (_) | | | \__ \     |
|       \____|_|\__|_| |_|\__,_|_.__/  /_/   \_\___|\__|_|\___/|_| |_|___/     |
|                                                                              |
|                       Self-hosted runner registration                        |
|                                                                              |
--------------------------------------------------------------------------------

# Authentication


√ Connected to GitHub

# Runner Registration




√ Runner successfully added
√ Runner connection is good

# Runner settings


√ Settings Saved.

2022-10-20 02:19:36.697  DEBUG --- Runner successfully configured.
{
  "agentId": 771593,
  "agentName": "gha-self-hosted-runner-4kp8x-f426w",
  "poolId": 1,
  "poolName": "Default",
  "disableUpdate": true,
  "ephemeral": true,
  "serverUrl": "https://pipelines.actions.githubusercontent.com/JF5L2KFQpX53qZmBmu2KdqcaCz4ICNJOuFwyQ9ZwJx7CmEPtrg",
  "gitHubUrl": "https://github.com/myorg",
  "workFolder": "/runner/_work"
2022-10-20 02:19:36.700  DEBUG --- Docker enabled runner detected and Docker daemon wait is enabled
2022-10-20 02:19:36.701  DEBUG --- Waiting until Docker is available or the timeout is reached
}CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
{
  "apiVersion": "actions.summerwind.dev/v1alpha1",
  "kind": "Runner",
  "metadata": {
    "annotations": {
      "cluster-autoscaler.kubernetes.io/safe-to-evict": "false",
      "iam.amazonaws.com/role": "lf_gha_assumer",
      "sync-time": "2022-10-20T02:11:00Z"
    },
    "creationTimestamp": "2022-10-20T02:11:00Z",
    "finalizers": [
      "runner.actions.summerwind.dev"
    ],
    "generateName": "gha-self-hosted-runner-4kp8x-",
    "generation": 1,
    "labels": {
      "app.kubernetes.io/name": "gha-self-hosted-runner",
      "disalow-privileged-containers/enable": "false",
      "runner-deployment-name": "gha-self-hosted-runner",
      "runner-template-hash": "6776bdd8db"
    },
    "managedFields": [
      {
        "apiVersion": "actions.summerwind.dev/v1alpha1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:metadata": {
            "f:annotations": {
              ".": {},
              "f:cluster-autoscaler.kubernetes.io/safe-to-evict": {},
              "f:iam.amazonaws.com/role": {},
              "f:sync-time": {}
            },
            "f:finalizers": {
              ".": {},
              "v:\"runner.actions.summerwind.dev\"": {}
            },
            "f:generateName": {},
            "f:labels": {
              ".": {},
              "f:app.kubernetes.io/name": {},
              "f:disalow-privileged-containers/enable": {},
              "f:runner-deployment-name": {},
              "f:runner-template-hash": {}
            },
            "f:ownerReferences": {
              ".": {},
              "k:{\"uid\":\"a035e991-1c56-4bf9-a387-b19252ca0d66\"}": {}
            }
          },
          "f:spec": {
            ".": {},
            "f:dockerVolumeMounts": {},
            "f:dockerdContainerResources": {
              ".": {},
              "f:limits": {
                ".": {},
                "f:memory": {}
              },
              "f:requests": {
                ".": {},
                "f:cpu": {},
                "f:memory": {}
              }
            },
            "f:dockerdWithinRunnerContainer": {},
            "f:env": {},
            "f:ephemeral": {},
            "f:image": {},
            "f:imagePullPolicy": {},
            "f:nodeSelector": {
              ".": {},
              "f:group-name": {}
            },
            "f:organization": {},
            "f:resources": {
              ".": {},
              "f:limits": {
                ".": {},
                "f:memory": {}
              },
              "f:requests": {
                ".": {},
                "f:cpu": {},
                "f:memory": {}
              }
            },
            "f:serviceAccountName": {},
            "f:tolerations": {},
            "f:topologySpreadConstraints": {},
            "f:volumeMounts": {},
            "f:volumes": {}
          }
        },
        "manager": "manager",
        "operation": "Update",
        "time": "2022-10-20T02:11:00Z"
      },
      {
        "apiVersion": "actions.summerwind.dev/v1alpha1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:status": {
            ".": {},
            "f:registration": {
              ".": {},
              "f:expiresAt": {},
              "f:organization": {},
              "f:token": {}
            }
          }
        },
        "manager": "manager",
        "operation": "Update",
        "subresource": "status",
        "time": "2022-10-20T02:11:00Z"
      },
      {
        "apiVersion": "actions.summerwind.dev/v1alpha1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:status": {
            "f:message": {},
            "f:phase": {}
          }
        },
        "manager": "curl",
        "operation": "Update",
        "subresource": "status",
        "time": "2022-10-20T02:13:45Z"
      }
    ],
    "name": "gha-self-hosted-runner-4kp8x-f426w",
    "namespace": "ci-runners",
    "ownerReferences": [
      {
        "apiVersion": "actions.summerwind.dev/v1alpha1",
        "blockOwnerDeletion": true,
        "controller": true,
        "kind": "RunnerReplicaSet",
        "name": "gha-self-hosted-runner-4kp8x",
        "uid": "a035e991-1c56-4bf9-a387-b19252ca0d66"
      }
    ],
    "resourceVersion": "929416370",
    "uid": "a685e75e-56e5-403c-94cf-d9eec37cbc5c"
  },
  "spec": {
    "dockerVolumeMounts": [
      {
        "mountPath": "/root/.docker/config.json",
        "name": "docker-login",
        "subPath": "config.json"
      }
    ],
    "dockerdContainerResources": {
      "limits": {
        "memory": "2Gi"
      },
      "requests": {
        "cpu": "150m",
        "memory": "256Mi"
      }
    },
    "dockerdWithinRunnerContainer": false,
    "env": [
      {
        "name": "DISABLE_RUNNER_UPDATE",
        "value": "true"
      }
      {
        "name": "LOG_DEBUG_DISABLED",
        "value": "true"
      }
    ],
    "ephemeral": true,
    "image": "summerwind/actions-runner:v2.298.2-ubuntu-20.04",
    "imagePullPolicy": "IfNotPresent",
    "nodeSelector": {
      "group-name": "ci-general-group"
    },
    "organization": "myorg",
    "resources": {
      "limits": {
        "memory": "2Gi"
      },
      "requests": {
        "cpu": "200m",
        "memory": "128Mi"
      }
    },
    "serviceAccountName": "github-action",
    "tolerations": [
      {
        "effect": "NoSchedule",
        "key": "node-role.kubernetes.io/ci-general"
      }
    ],
    "topologySpreadConstraints": [
      {
        "labelSelector": {
          "matchLabels": {
            "app.kubernetes.io/name": "gha-self-hosted-runner"
          }
        },
        "maxSkew": 1,
        "topologyKey": "kubernetes.io/hostname",
        "whenUnsatisfiable": "ScheduleAnyway"
      }
    ],
    "volumeMounts": [
      {
        "mountPath": "/home/runner/.docker/config.json",
        "name": "docker-login",
        "subPath": "config.json"
      }
    ],
    "volumes": [
      {
        "name": "docker-login",
        "secret": {
          "items": [
            {
              "key": ".dockerconfigjson",
              "path": "config.json"
            }
          ],
          "secretName": "docker-login"
        }
      }
    ]
  },
  "status": {
    "message": "",
    "phase": "Idle",
    "registration": {
      "expiresAt": "2022-10-20T03:00:45Z",
      "organization": "myorg",
      "token": "OMIT*****KC453AVPNFXHG5DBNRWGC5DJN5XF62LEZYAQHIKTWFUW443UMFWGYYLUNFXW4X3UPFYGLN2JNZ2GKZ3SMF2GS33OJFXHG5DBNRWGC5DJN5XA"
    }
  }
}
√ Connected to GitHub

Current runner version: '2.298.2'
2022-10-20 02:19:37Z: Listening for Jobs
```

**Comment by mumoshu** [comment]
Good catch! Apparently, we do need to fix this to correctly support the envvars "suggested" in the comment within the logger.bash. Would you mind contributing it?

**Comment by yuriydzobak** [comment]
> Thanks for reporting. Good catch! Apparently, we do need to fix this to correctly support the envvars "suggested" in the comment within the logger.bash. Would you mind contributing it?

I can't understand where is problem, I checked locally, I can't solve problem
I think it should be removed all `exec` in [LINK](https://github.com/actions-runner-controller/actions-runner-controller/pull/1268/files#diff-6d715b4ccae8f901769c62fe11c216e49e7b21f0a68a4199239a7bf26d5e07b2)

**Comment by mumoshu** [comment]
FYI, you can use our Makefile to build a custom runner image
https://github.com/actions-runner-controller/actions-runner-controller/blob/master/CONTRIBUTING.md#developing-the-controller

**Comment by mumoshu** [comment]
It turned out this isn't related to LOG_DBEUG_DISABLED as explained in https://github.com/actions-runner-controller/actions-runner-controller/pull/2000.
But the original issue of dumping the whole runner pod manifest including spec and status to stdout has already been fixed via #1759, by updating the redirection at https://github.com/actions-runner-controller/actions-runner-controller/blame/master/runner/update-status#L30

**Comment by mumoshu** [comment]
Closing as resolved
